In [1]:
import numpy as np
import pandas as pd

In [2]:
# The statistical module used to run chi square test
import scipy.stats as stats

In [3]:
# Observed data in a (hypothetical) survey of 6000 people 
observed = pd.Series([1500, 1350, 1600, 1550], index=["1", "2", "3", "4"])

# Create a data frame
df = pd.DataFrame([observed]).T

# Add a column whose default values are the expected values
df[1] = 1500

# Rename the columns
df.columns = ["observed", "expected"]

# View the data frame
df

observed  expected
1      1500      1500
2      1350      1500
3      1600      1500
4      1550      1500

In [4]:
# With four rows, the degree of freedom is 4-1 = 3
# With a p-value of 0.05, the confidence level is 1.00-0.05 = 0.95.
critical_value = stats.chi2.ppf(q = 0.95, df = 3)

In [5]:
# The critical value
critical_value

7.8147279032511765

In [6]:
# Run the chi square test with stats.chisquare()
stats.chisquare(df['observed'], df['expected'])

Power_divergenceResult(statistic=23.333333333333336, pvalue=3.4411276663397364e-05)

### Conclusion
* Since the chi-square value of 23.3 at a confidence level of 95% exceeds the critical value of 7.8, we conclude that the differences seen in the number of visits per cafe are statistically significant.